File: base/test/android/javatests/src/org/chromium/base/test/util/IntegrationTest.java
// Copyright 2014 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.base.test.util;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * This annotation is for integration tests.
 * <p>
 * Examples of integration tests are tests that rely on real instances of the
 * application's services and components (e.g. Search) to test the system as
 * a whole. These tests may use additional command-line flags to configure the
 * existing backends to use.
 * <p>
 * Such tests are likely NOT reliable enough to run on tree closing bots and
 * should only be run on FYI bots.
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface IntegrationTest {
}

File: packages/integration_test/example/test_driver/integration_test.dart
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter_driver/flutter_driver.dart';
import 'package:integration_test/integration_test_driver.dart';

Future<void> main(List<String> args) async {
  final String testOutputDirectory = getTestOutputDirectory(args);
  integrationDriver(writeResponseOnFailure: true, testOutputDirectory: testOutputDirectory);
}
